tools/discord_notifier.py

```
@@ -0,0 +1,159 @@
+
+def notify_discord(webhook_url: str, content: str) -> bool:
+    """Send message to Discord webhook."""
+    try:
+        timeout = httpx.Timeout(connect=10.0, read=20.0, write=10.0, pool=30.0)
+        with httpx.Client(timeout=timeout) as client:
+            response = client.post(webhook_url, json={"content": content})
+            response.raise_for_status()
+            logger.info("Discord notification sent successfully")
+            return True
+    except httpx.HTTPStatusError as e:
+        logger.error(f"Discord webhook failed: {e.response.text}")
```

Ruff (0.12.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)